packages/cdc-react/src/components/ProfileHeader/ProfileHeader.tsx

```diff
@@ -24,34 +24,147 @@ export interface ProfileHeaderNotification {
 export const ProfileHeader = ({
   logo,
   className,
-  profileHeaderPopupOpen,
-  menuItems,
+  menuItems = [],
   popupMenuItems,
 }: ProfileHeaderProps) => {
+  const [profileHeaderPopupOpen, setProfileHeaderPopupOpen] = useState(false);
+
+  const popupRef = useRef<HTMLDivElement>(null);
+
+  const handleClickOutsidePopup: EventListenerOrEventListenerObject = (
+    ev: Event
+  ) => {
+    ev.preventDefault();
+
+    if (popupRef.current && !popupRef.current?.contains(ev.target as Node)) {
+      setProfileHeaderPopupOpen(false);
+    }
+
+    const closestButton: HTMLButtonElement | null = (
+      ev.target as HTMLElement
+    ).closest("button");
+
+    const closestButtonIsUserProfileMenuItem: boolean =
+      (closestButton &&
+        closestButton.classList.contains(
+          "profile-header-user-profile-menu-item"
+        )) ||
+      false;
+
+    if (profileHeaderPopupOpen === true && closestButtonIsUserProfileMenuItem) {
+      // Added setTimeout to prevent issues inside the performance events timing cycle
+      setProfileHeaderPopupOpen(false);
+    } else if (
+      profileHeaderPopupOpen === false &&
+      closestButtonIsUserProfileMenuItem
+    ) {
+      setProfileHeaderPopupOpen(true);
+    }
+  };
+
+  const handleKeyDown = function (ev: React.KeyboardEvent) {
```

Comment:
Consider using `tabindex` to utilize built-in HTML/DOM features.
https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex